source/agora/consensus/SlashPolicy.d

```diff
@@ -113,6 +113,36 @@ public class SlashPolicy
         }
     }
 
+    /***************************************************************************
+
+        Get the UTXOs of the validators that do not reveal their pre-images
+        by indices
+
+        Params:
+            validators_utxos = will contain the UTXOs ot the validators
+            missing_validators = indices of validators being slashed
+
+    ***************************************************************************/
+
+    public void getMissingValidatorsUTXOs (ref Hash[] validators_utxos,
+        const ref uint[] missing_validators) @safe nothrow
+    {
+        validators_utxos.length = 0;
+        () @trusted { assumeSafeAppend(validators_utxos); }();
+
+        Hash[] keys;
+        if (!this.enroll_man.getEnrolledUTXOs(keys))
+        {
+            log.fatal("Could not retrieve enrollments");
+            assert(0);
+        }
+
+        foreach (idx; missing_validators)
+        {
```

Comment:
This isn't actually doing anything? The array is empty.

Reply:
> This isn't actually doing anything? The array is empty.

The `missing_validators` which is in fact `missing_validators` in `BlockHeader` is passed from the Ledger when creating a new Block. Can you explain why do you think it's empty?

Reply:
ahh I misread the code. I saw the other array set to zero.